book/spiders/books.py

```diff
@@ -0,0 +1,43 @@
+import scrapy
+from scrapy.http.response import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs: dict) -> None:
+        for book in response.css(".product_pod"):
+            book_page = response.urljoin(
+                book.css(".product_pod h3 a::attr(href)").get()
+            )
+            yield scrapy.Request(book_page, callback=self.parse_single_book)
+
+        next_page = response.css("li.next a::attr(href)").get()
+
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
+
+    @staticmethod
+    def parse_single_book(book_page: Response) -> dict[str, str]:
+        detail_info = {
+            "title": book_page.css(".product_main > h1::text").get(),
+            "price": book_page.css(".price_color::text").get(),
+            "amount_in_stock": (
+                book_page.css("tr:nth-child(6) > td::text").get()
+                .replace("(", "")
+                .split(" ")[2]
```

Comment:
Maybe it would be better to display the amount in stock as an integer

Reply:
not required but make some sense